core/domain/collection_services.py

```diff
@@ -372,10 +477,15 @@ def get_collection_ids_matching_query(query_string, cursor=None):
     """Returns a list with all collection ids matching the given search query
     string, as well as a search cursor for future fetches.
 
-    This method returns exactly feconf.SEARCH_RESULTS_PAGE_SIZE results if
-    there are at least that many, otherwise it returns all remaining results.
-    (If this behaviour does not occur, an error will be logged.) The method
-    also returns a search cursor.
+    Args:
+        query_string: str. The search query string.
+
+    Returns:
+        list(str). A list with all collection ids matching the given search
```

Comment:
I don't think this is clear, it needs to be elaborated more. I mean, it's not a list(str) -- it's a list with two elements, the first of which is a list and the second of which is a str.

There's a good example here of how to explain something like this: https://github.com/oppia/oppia/pull/2465/files#diff-e8e75b25b618133593df5ff9cd67a59bR228